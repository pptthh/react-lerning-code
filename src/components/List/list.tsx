import * as React from 'react';
import ListItem, { IListItem } from './listItem';

export interface IList {
    list: IListItem[];
}

const List = ({list}: IList): JSX.Element =>
<ul className='list'>
    {list.map((item: IListItem) => <ListItem {...item} key={item.time}/>)}
</ul>;

export default List;
