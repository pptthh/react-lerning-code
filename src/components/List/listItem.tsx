import * as React from 'react';

export interface IListItem {
    text: JSX.Element | string;
    time: number;
}

const ListItem = ({time, text}: IListItem) => <li className='ListItem'>{text}</li>;

export default ListItem;
