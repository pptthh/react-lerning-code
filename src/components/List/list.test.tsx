import { configure, render } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import List from './list';

configure({adapter: new ReactSixteenAdapter()});

it('renders correctly List with correct data', () => {
    const LIST_TEST = [
        {
            text: 'test text',
            time: 0,
        },
        {
            text: 'test text2',
            time: 1111111111111,
        },
    ];

    const tree = render(
        <List list={LIST_TEST}/>);
    expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly List with Empty Array', () => {
    const tree = render(<List list={[]}/>);
    expect(toJson(tree)).toMatchSnapshot();
});
