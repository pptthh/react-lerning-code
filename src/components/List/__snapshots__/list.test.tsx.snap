// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly List with Empty Array 1`] = `
<ul
  class="list"
/>
`;

exports[`renders correctly List with correct data 1`] = `
<ul
  class="list"
>
  <li
    class="ListItem"
  >
    test text
  </li>
  <li
    class="ListItem"
  >
    test text2
  </li>
</ul>
`;
